data-loader/cli/src/main/java/com/scalar/db/dataloader/cli/command/dataexport/ExportCommand.java

```diff
@@ -126,11 +131,100 @@ private void validateOutputDirectory() throws DirectoryValidationException {
     }
   }
 
+  /**
+   * Creates a {@link TableMetadataService} instance based on the specified {@link ScalarDbMode} and
+   * ScalarDB configuration file.
+   *
+   * <p>If the mode is {@code TRANSACTION}, this method initializes a {@link TransactionFactory} and
+   * uses its transaction admin to create a {@link TableMetadataTransactionService}. Otherwise, it
+   * initializes a {@link StorageFactory} and creates a {@link TableMetadataStorageService} using
+   * its storage admin.
+   *
+   * @param scalarDbMode the mode ScalarDB is running in (either {@code STORAGE} or {@code
+   *     TRANSACTION})
+   * @param scalarDbPropertiesFilePath the path to the ScalarDB properties file
+   * @return an appropriate {@link TableMetadataService} based on the mode
+   * @throws IOException if reading the ScalarDB properties file fails
+   */
+  private TableMetadataService createTableMetadataService(
+      ScalarDbMode scalarDbMode, String scalarDbPropertiesFilePath) throws IOException {
+    if (scalarDbMode.equals(ScalarDbMode.TRANSACTION)) {
+      TransactionFactory transactionFactory = TransactionFactory.create(scalarDbPropertiesFilePath);
+      return new TableMetadataTransactionService(transactionFactory.getTransactionAdmin());
+    }
+    StorageFactory storageFactory = StorageFactory.create(scalarDbPropertiesFilePath);
+    return new TableMetadataStorageService(storageFactory.getStorageAdmin());
+  }
+
+  /**
+   * Creates an {@link ExportManager} instance based on ScalarDB mode and file format.
+   *
+   * @param scalarDbMode The ScalarDB mode (TRANSACTION or STORAGE).
+   * @param scalarDbDao The DAO for accessing ScalarDB.
+   * @param fileFormat The output file format (CSV, JSON, JSONL).
+   * @param scalarDbPropertiesFilePath Path to the ScalarDB properties file.
+   * @return A configured {@link ExportManager}.
+   * @throws IOException If there is an error reading the properties file.
+   */
   private ExportManager createExportManager(
-      StorageFactory storageFactory, ScalarDbDao scalarDbDao, FileFormat fileFormat) {
+      ScalarDbMode scalarDbMode,
+      ScalarDbDao scalarDbDao,
+      FileFormat fileFormat,
+      String scalarDbPropertiesFilePath)
+      throws IOException {
     ProducerTaskFactory taskFactory =
         new ProducerTaskFactory(delimiter, includeTransactionMetadata, prettyPrintJson);
-    DistributedStorage storage = storageFactory.getStorage();
+    if (scalarDbMode.equals(ScalarDbMode.STORAGE)) {
+      DistributedStorage storage = StorageFactory.create(scalarDbPropertiesFilePath).getStorage();
+      return createExportManagerWithStorage(storage, scalarDbDao, fileFormat, taskFactory);
+    } else {
+      DistributedTransactionManager distributedTransactionManager =
```

Comment:
Don't need to consider https://github.com/scalar-labs/scalardb/pull/2740#issuecomment-2982959912 ?